microbenchmarks/results.txt

```diff
@@ -0,0 +1,89 @@
+DigestTypeBenchmark.digestManager:·gc.churn.G1_Eden_Space                 NOT_ARRAY_BACKED       MAC  thrpt    5    539.102 ±    0.215  MB/sec
+DigestTypeBenchmark.digestManager:·gc.churn.G1_Eden_Space.norm            NOT_ARRAY_BACKED       MAC  thrpt    5  70401.331 ±  403.899    B/op
+DigestTypeBenchmark.digestManager:·gc.churn.G1_Old_Gen                    NOT_ARRAY_BACKED       MAC  thrpt    5      0.006 ±    0.015  MB/sec
+DigestTypeBenchmark.digestManager:·gc.churn.G1_Old_Gen.norm               NOT_ARRAY_BACKED       MAC  thrpt    5      0.784 ±    1.962    B/op
+DigestTypeBenchmark.digestManager:·gc.count                               NOT_ARRAY_BACKED       MAC  thrpt    5     55.000             counts
+DigestTypeBenchmark.digestManager:·gc.time                                NOT_ARRAY_BACKED       MAC  thrpt    5    200.000                 ms
+DigestTypeBenchmark.digestManager:·stack                                  NOT_ARRAY_BACKED       MAC  thrpt             NaN                ---
+DigestTypeBenchmark.digestManager                                         NOT_ARRAY_BACKED     CRC32  thrpt    5    119.933 ±    7.129  ops/ms
+DigestTypeBenchmark.digestManager:·gc.alloc.rate                          NOT_ARRAY_BACKED     CRC32  thrpt    5   7206.443 ±  430.557  MB/sec
+DigestTypeBenchmark.digestManager:·gc.alloc.rate.norm                     NOT_ARRAY_BACKED     CRC32  thrpt    5  65648.001 ±    0.001    B/op
+DigestTypeBenchmark.digestManager:·gc.churn.G1_Eden_Space                 NOT_ARRAY_BACKED     CRC32  thrpt    5   7684.923 ±  509.880  MB/sec
+DigestTypeBenchmark.digestManager:·gc.churn.G1_Eden_Space.norm            NOT_ARRAY_BACKED     CRC32  thrpt    5  70005.330 ±  532.187    B/op
+DigestTypeBenchmark.digestManager:·gc.churn.G1_Old_Gen                    NOT_ARRAY_BACKED     CRC32  thrpt    5      0.158 ±    0.078  MB/sec
+DigestTypeBenchmark.digestManager:·gc.churn.G1_Old_Gen.norm               NOT_ARRAY_BACKED     CRC32  thrpt    5      1.439 ±    0.769    B/op
+DigestTypeBenchmark.digestManager:·gc.count                               NOT_ARRAY_BACKED     CRC32  thrpt    5    784.000             counts
+DigestTypeBenchmark.digestManager:·gc.time                                NOT_ARRAY_BACKED     CRC32  thrpt    5   2949.000                 ms
+DigestTypeBenchmark.digestManager:·stack                                  NOT_ARRAY_BACKED     CRC32  thrpt             NaN                ---
+DigestTypeBenchmark.digestManager                                         NOT_ARRAY_BACKED   CRC32_C  thrpt    5    130.444 ±    3.441  ops/ms
+DigestTypeBenchmark.digestManager:·gc.alloc.rate                          NOT_ARRAY_BACKED   CRC32_C  thrpt    5   7824.727 ±  212.028  MB/sec
+DigestTypeBenchmark.digestManager:·gc.alloc.rate.norm                     NOT_ARRAY_BACKED   CRC32_C  thrpt    5  65552.001 ±    0.001    B/op
```

Comment:
Having 64Kb allocated (and GCed) per each buffer seems to indicate that something is off. Prob, that's because of the pure java implementation fallback due to not having the JNI lib on MacOS.

Reply:
@merlimat in unit tests I did not see this log: 
        if (!Sse42Crc32C.isSupported()) {
            LOG.error("Sse42Crc32C is not supported, will use less slower CRC32C implementation.");
        }
so I assume that it won't use pure java implementation.

This is because DoubleByteBuf returns  false == hasMemoryAddress() and false == hasArray()

    public static int computeChecksum(ByteBuf payload) {
        if (payload.hasMemoryAddress() && (CRC32C_HASH instanceof Sse42Crc32C)) {
            return CRC32C_HASH.calculate(payload.memoryAddress() + payload.readerIndex(), payload.readableBytes());
        } else if (payload.hasArray()) {
            return CRC32C_HASH.calculate(payload.array(), payload.arrayOffset() + payload.readerIndex(),
                    payload.readableBytes());
        } else {
            return CRC32C_HASH.calculate(payload.nioBuffer());
        }
    }

ends up in .nioBuffer() which ends up doing the following:

    public ByteBuffer nioBuffer(int index, int length) {
      >>>>>   ByteBuffer dst = ByteBuffer.allocate(length);
        ByteBuf b = Unpooled.wrappedBuffer(dst);

Reply:
Oh, that's true. That's because it's doing that on a `DoubleByteBuf`. We should make sure we don't do that in the real code :)